### PR TITLE
Enochtsang/send to rabbitmq

### PR DESCRIPTION
**Pull request by enochtsang**

### CODE REVIEW

**Comment by enochtsang** [comment]
@JordanHeinrichs please review

File: src/BusinessLayer/I_CommunicationService.h [hunk 1]
```diff
@@ -11,7 +11,7 @@ class I_CommunicationService : public QObject
     virtual ~I_CommunicationService() {}
 
     virtual void sendSerialData(const unsigned char* packet, int packetLength) = 0;
-    virtual void sendInternetData(const QByteArray& data) = 0;
+    virtual void sendInternetData(const QString& data) = 0;
```
Comment (JordanHeinrichs) [inline src/BusinessLayer/I_CommunicationService.h]: What was the reason for switching to QString? The JSON data is originally coming from QByteArray and it is going to std string (which is also a function in QByteArray). At a glance it seems like this has an unneeded conversion.

File: src/BusinessLayer/CommunicationService.cpp [hunk 2]
```diff
@@ -1,54 +1,42 @@
...
-    view_.setConnectionStatus(serialPeripheral_->attemptConnection());
+    serialView_.setConnectionStatus(serialPeripheral_.attemptConnection(serialView_.getCommunicationPort()));
 }
 
 void CommunicationService::attemptInternetConnection()
 {
-    //TODO: Implement or replace this function to try to connect to the server
+    internetView_.setConnectionStatus(internetPeripheral_.attemptConnection(
+                                          internetView_.getIpAddress(), internetView_.getPort()));
+    qDebug() << "CommunicationService::attemptInternetConnection";
```
Comment (JordanHeinrichs) [inline src/BusinessLayer/CommunicationService.cpp]: This debug statement could be removed. Or optionally I think it should be the first thing within the function if you still want to keep it (which is fine as there is not a whole lot of feedback when hitting connect).

Comment (enochtsang) [inline src/BusinessLayer/CommunicationService.cpp]: Yup extraneous, was using it for debugging the seg fault.

File: src/BusinessLayer/I_CommunicationService.h (hunk 1 above)
Comment (enochtsang) [inline src/BusinessLayer/I_CommunicationService.h]: True

Comment (enochtsang) [inline src/BusinessLayer/I_CommunicationService.h]: True

**Pull request merged**